src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
             return node.CreateBindingSuccess(value);
         }
 
-        return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        var bindingResult = MatchExpressionValue(node, expression);
+        if ( bindingResult is not null)
+        {
+            return bindingResult;
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        }
+
+        
     }
+
+    private HttpBindingResult<object?>? MatchExpressionValue(HttpExpressionNode node, string expression)
+    {
+
+        var guidPattern = new Regex(@$"\{"$guid"}", RegexOptions.Compiled);
+        var dateTimePattern = new Regex(@$"\{"$datetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+
+        var guidMatches = guidPattern.Matches(expression);
+        var dateTimeMatches = dateTimePattern.Matches(expression);
+        var localDateTimeMatches = localDateTimePattern.Matches(expression);
+        var randomIntMatches = randomIntPattern.Matches(expression);
+        var timestampMatches = timestampPattern.Matches(expression);
+
+        if (guidMatches.Count > 0)
+        {
+            return node.CreateBindingSuccess(Guid.NewGuid().ToString());
+        } else if(dateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, dateTimeMatches, dateTimePattern);
+        } else if(localDateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, localDateTimeMatches, localDateTimePattern);
+        } else if(randomIntMatches.Count > 0)
+        {
+            return GetRandInt(node, expression, randomIntMatches);
+        }else if(timestampMatches.Count > 0)
+        {
+            return GetTimestamp(node, expression, timestampMatches);
+        }
+        return null;
```

Comment:
consider: It may be better if this function also returned the `node.CreateBindingFailure(HttpDiagnostics.UnableToEvaluateExpression(expression));` instead of `null`. That way you could make the return type non-nullable and the caller can just say `return MatchExpressionValue(node, expression);`

Reply:
I think returning `null` here feels all the more odd considering that the called `Get*` functions already always return a non-nullable result that can be either success or failure.